src/main/java/org/craftercms/engine/servlet/filter/SiteAwareCORSFilter.java

```diff
@@ -0,0 +1,70 @@
+package org.craftercms.engine.servlet.filter;
+
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletResponseWrapper;
+
+import org.apache.commons.configuration2.HierarchicalConfiguration;
+import org.craftercms.commons.web.CORSFilter;
+import org.craftercms.engine.service.context.SiteContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/**
+ * Filter to include CORS headers based on the current site configuration.
+ *
+ * @author Jose Ross
+ */
+public class SiteAwareCORSFilter extends OncePerRequestFilter {
+
+    private static final Logger logger = LoggerFactory.getLogger(SiteAwareCORSFilter.class);
+
+    public static final String CONFIG_KEY = "cors";
+    public static final String ENABLE_KEY = "[@enable]";
+    public static final String MAX_AGE_KEY = "accessControlMaxAge";
+    public static final String ALLOW_ORIGIN_KEY = "accessControlAllowOrigin";
+    public static final String ALLOW_METHODS_KEY = "accessControlAllowMethods";
+    public static final String ALLOW_HEADERS_KEY = "accessControlAllowHeaders";
+    public static final String ALLOW_CREDENTIALS_KEY = "accessControlAllowCredentials";
+
+    /**
+     * Exclude requests for sites that have no configuration or it is marked as disabled.
+     */
+    @Override
+    protected boolean shouldNotFilter(final HttpServletRequest request) throws ServletException {
+        SiteContext siteContext = SiteContext.getCurrent();
+        HierarchicalConfiguration config = siteContext.getConfig();
+        try {
+            HierarchicalConfiguration corsConfig = config.configurationAt(CONFIG_KEY);
+            if (corsConfig != null) {
+                boolean enable = corsConfig.getBoolean(ENABLE_KEY, false);
```

Comment:
Just return the result of corsConfig.getBoolean, instead of using a variable and then immediately returning it. The variable is not necessary and I think even Codacy wil complain.